Sources/GeneratorEngine/Engine.swift

```diff
@@ -27,12 +28,10 @@ public func withEngine(
     cacheLocation: cacheLocation
   )
 
-  do {
+  try await withAsyncThrowingDefer {
     try await body(engine)
+  } deferring: {
```

Comment:
WDYT of a slightly different naming?

```suggestion
  try await withAsyncThrowing {
    try await body(engine)
  } defer: {
```

Reply:
It seems to build. Xcode (SourceKit) will format the `defer` here like the actual `defer` keyword, but otherwise it seems to work.